### ISSUE TITLE
Data prep - calculate 3d keypoints

### ISSUE THREAD

**Issue by johndpope** [johndpope/MegaPortrait-hack#3]
Since we predict 3D instead of 2D keypoints, We use an orthographic projection to project the keypoints to the image plane before computing the loss.
https://ar5iv.labs.arxiv.org/html/2011.15126

![Screenshot from 2024-05-13 14-19-12](https://github.com/johndpope/MegaPortrait-hack/assets/289994/d4c063a1-e80c-4cc7-a9f7-e2e5b134ff48)

https://github.com/1adrianb/face-alignment
![Screenshot from 2024-05-13 21-34-09](https://github.com/johndpope/MegaPortrait-hack/assets/289994/60fa7bf9-6eb6-4c65-b3f9-c42dfcd7c93f)

**Comment by francqz31** [comment]
@johndpope have you tried implementing the paper using gpt4-o ? it looks better than opus

**Comment by johndpope** [comment]
Just playing now on iPhone 
I did through paper into prompt - it confidently spat out garbage. It might be great to hone in on aligning the diagrams to code (see reference folder - Claude got this wrong had to correct it)  - will test later. 

Also
I did update training code with a face alignment - 2d points - can adjust later 3d - I need to start debugging the training but looks much better than prior work. 


UPDATE
it seems to truncate file uploaded - can't handle the context window like Claude - 200k
waste of time.
![Screenshot from 2024-05-14 14-10-09](https://github.com/johndpope/MegaPortrait-hack/assets/289994/6f0656a8-51ad-4be8-b816-0fbdc7172df3)
